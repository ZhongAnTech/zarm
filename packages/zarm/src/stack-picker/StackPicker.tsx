import React, {
  useEffect,
  MouseEvent,
  useState,
  ReactNode,
  useCallback,
  memo,
  useMemo,
} from 'react';
import classnames from 'classnames';
import Popup from '../popup';
import Actions from './Actions';
import PropsType, { TDataSource, IDataSource } from './PropsType';

const stopPropagation = (e: MouseEvent) => {
  e.stopPropagation();
};

export interface StackPickerProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default memo(
  ({
    prefixCls = 'za-stack-picker',
    className,
    dataSource,
    defaultValue,
    value,
    displayMember = 'label',
    valueMember = 'value',
    title = '请选择',
    visible = false,
    cols = Infinity,
    labelAddon = ' > ',
    displayRender,
    itemRender = (data: TDataSource) => data[displayMember as 'label'],
    onChangeValidate,
    cancelText,
    okText,
    onChange,
    onOk,
    onCancel,
    maskClosable = false,
    locale,
  }: StackPickerProps) => {
    const [errorMsg, setErrorMsg] = useState<ReactNode>('');
    const [currentValue, setCurrentValue] = useState<IDataSource[]>([]);
    const [currentVisible, setCurrentVisible] = useState(visible);

    const handleObtainItem = useCallback(
      (list: typeof dataSource, _value: ReactNode) => {
        return list.find((item) => item[valueMember as 'value'] === _value);
      },
      [valueMember],
    );

    useEffect(() => {
      setCurrentVisible(visible);
    }, [visible]);

    /**
     * 初始化列表值
     */
    useEffect(() => {
      const initValue: typeof dataSource = [];
      const v = value || defaultValue || [];

      v.reduce((accumulator: IDataSource[], _currentValue: ReactNode) => {
        const valueItem = handleObtainItem(accumulator, _currentValue);

        if (valueItem) {
          initValue.push(valueItem);

          return valueItem.children || [];
        }

        return [];
      }, dataSource);

      setCurrentValue(initValue);
    }, [dataSource, handleObtainItem, value, defaultValue]);

    /**
     * 点击遮罩
     */
    const handleMaskClick = () => {
      if (!maskClosable) return;

      if (onCancel) {
        if (typeof onCancel !== 'function') {
          console.error('onCancel need a function');
        } else {
          onCancel();
        }
      }

      setCurrentVisible(!currentVisible);
    };

    /**
     * 点击取消按钮
     */
    const handleCancel = () => {
      if (onCancel) {
        if (typeof onCancel !== 'function') {
          console.error('onCancel need a function');
        } else {
          onCancel();
        }
      }
    };

    /**
     * 获取列表数据
     */
    const columnDataList = useMemo(() => {
      const group: typeof dataSource[] = [];
      let _dataSource: typeof dataSource | null = dataSource;
      let i = 0;
      while (_dataSource) {
        group.push(_dataSource);

        const colVal = currentValue[i];
        const childrenData: typeof dataSource | undefined = (
          (colVal
            ? handleObtainItem(_dataSource, colVal[valueMember as keyof ReactNode])
            : _dataSource[0]) || {}
        ).children;

        if (childrenData && childrenData.length && i < cols - 1) {
          _dataSource = childrenData;
        } else {
          _dataSource = null;
        }

        i += 1;
      }

      return group;
    }, [cols, currentValue, dataSource, handleObtainItem, valueMember]);

    /**
     * 修改列表值
     * @param e
     * @param index
     * @param item
     */
    const handleChange = (e: MouseEvent, index?: number, item?: IDataSource) => {
      stopPropagation(e);

      const _value = currentValue.slice(0, index);

      if (item && typeof index === 'number') {
        _value[index] = item;
      }

      setCurrentValue(_value);

      if (onChangeValidate) {
        if (typeof onChangeValidate !== 'function') {
          console.error('onChangeValidate need a function');
        } else {
          const _errorMsg = onChangeValidate(value);
          setErrorMsg(_errorMsg);
        }
      }

      if (onChange) {
        if (typeof onChange !== 'function') {
          console.error('onChange need a function');
        } else {
          onChange(_value.map((v) => v![valueMember as 'value']));
        }
      }
    };

    /**
     * 点击确认按钮
     */
    const handleOk = () => {
      if (onOk) {
        if (typeof onOk !== 'function') {
          console.error('onOk need a function');
        } else {
          onOk(currentValue.map((v) => v![valueMember as 'value']));
        }
      }
    };

    /**
     * 判断是否有传入 itemRender， 否则使用默认的渲染方式
     * @param item
     */
    const handleItemRender = (item: TDataSource) => {
      return itemRender(item);
    };

    /**
     * 判断是否有传入 displayRender， 否则使用默认的渲染方式展示选择内容
     */
    const handleDisplayRender = () => {
      const _currentValue = currentValue.map(({ children, ...others }) => others);

      if (displayRender) {
        if (typeof displayRender !== 'function') {
          console.error('displayRender need a function');

          // 因为报错返回一个空字符串
          return '';
        }

        return displayRender(_currentValue);
      }

      return _currentValue.map((item) => handleItemRender(item)).join(labelAddon);
    };

    return (
      <Popup visible={currentVisible} onMaskClick={handleMaskClick} direction="bottom">
        <div className={classnames(prefixCls, className)}>
          <div className={`${prefixCls}__container`}>
            <div className={`${prefixCls}__wrapper`}>
              <Actions
                {...{
                  prefixCls,
                  cancelText,
                  okText,
                  title,
                  handleCancel,
                  handleOk,
                  locale,
                }}
              />

              <div className={`${prefixCls}__crumbs`}>
                <p>选择：{handleDisplayRender()}</p>

                {errorMsg && <p className={`${prefixCls}__crumbs-error`}>{errorMsg}</p>}
              </div>

              <div className={`${prefixCls}__group`}>
                {columnDataList.map((group, index) => {
                  const columnClass = classnames(`${prefixCls}__stack-column`, {
                    [`${prefixCls}__stack-column--hidden`]:
                      !currentValue[index] || !currentValue[index][valueMember as 'value'],
                  });

                  return (
                    <div
                      className={columnClass}
                      key={+index}
                      onClick={(e: MouseEvent<HTMLDivElement>) => handleChange(e, index - 1)}
                    >
                      <div
                        className={`${prefixCls}__stack-column-wrapper`}
                        onClick={(e) => stopPropagation(e)}
                      >
                        {group.map((item, i) => {
                          const isActive =
                            currentValue[index] &&
                            currentValue[index][valueMember as 'value'] ===
                              item[valueMember as 'value'];

                          const columnItemClass = classnames(`${prefixCls}__stack-column-item`, {
                            [`${prefixCls}__stack-column-item--active`]: isActive,
                          });

                          const label = handleItemRender(item);

                          return (
                            <div
                              key={`${label}${+i}`}
                              className={columnItemClass}
                              onClick={(e) => handleChange(e, index, item)}
                            >
                              {label}
                            </div>
                          );
                        })}
                      </div>
                    </div>
                  );
                })}
              </div>
            </div>
          </div>
        </div>
      </Popup>
    );
  },
);
