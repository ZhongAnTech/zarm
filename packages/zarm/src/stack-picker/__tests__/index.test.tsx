import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import StackPicker from '../index';

jest.useFakeTimers();

const District = [
  {
    value: '340000',
    label: '安徽省',
    children: [
      {
        value: '340800',
        label: '安庆市',
        children: [
          {
            value: '340803',
            label: '大观区',
            children: [],
          },
          {
            value: '340822',
            label: '怀宁县',
            children: [],
          },
          {
            value: '340882',
            label: '其它区',
            children: [],
          },
        ],
      },
    ],
  },
  {
    value: '310000',
    label: '上海',
    children: [
      {
        value: '310100',
        label: '上海市',
        children: [
          {
            value: '310113',
            label: '宝山区',
            children: [],
          },
          {
            value: '310105',
            label: '长宁区',
            children: [],
          },
          {
            value: '310230',
            label: '崇明县',
            children: [],
          },
          {
            value: '310152',
            label: '川沙区',
            children: [],
          },
        ],
      },
    ],
  },
];

describe('StackPicker', () => {
  it('renders correctly', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <StackPicker
        defaultValue={[]}
        value={[]}
        cancelText="取消"
        okText="确定"
        visible
        valueMember="value"
        itemRender={(data) => data.label}
        dataSource={District}
        onChange={onChange}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('handle props click', () => {
    const onCancel = jest.fn();
    const onOk = jest.fn();

    const wrapper = mount(
      <StackPicker
        defaultValue={[]}
        cancelText="取消"
        okText="确定"
        maskClosable
        value={['340000', '340800', '340803']}
        dataSource={District}
        onCancel={onCancel}
        onOk={onOk}
      />,
    );

    wrapper.setProps({ visible: true });
    wrapper.update();

    wrapper.find('.za-stack-picker__submit').simulate('click');
    expect(onOk).toHaveBeenCalledWith(['340000', '340800', '340803']);

    wrapper.find('.za-stack-picker__cancel').simulate('click');
    expect(onCancel).toBeCalled();

    wrapper.find('.za-mask').simulate('click');
    expect(onCancel).toBeCalled();
  });

  it('handle onChange', () => {
    const onChange = jest.fn();

    const wrapper = mount(
      <StackPicker
        defaultValue={[]}
        value={[]}
        cancelText="取消"
        okText="确定"
        visible
        dataSource={District}
        onChange={onChange}
      />,
    );

    // popupWrapper = mount(wrapper.instance().getComponent());

    wrapper
      .find('.za-stack-picker__group')
      .at(0)
      .find('.za-stack-picker__stack-column')
      .at(0)
      .simulate('click');

    wrapper
      .find('.za-stack-picker__group')
      .at(0)
      .find('.za-stack-picker__stack-column-wrapper')
      .at(0)
      .simulate('click');

    wrapper
      .find('.za-stack-picker__group')
      .at(0)
      .find('.za-stack-picker__stack-column-wrapper')
      .at(0)
      .find('.za-stack-picker__stack-column-item')
      .at(0)
      .simulate('click');

    expect(onChange).toHaveBeenCalledWith(['340000']);
  });

  it('handle onChangeValidate', () => {
    const onChange = jest.fn();
    const onChangeValidate = jest.fn(() => 'error');

    const wrapper = mount(
      <StackPicker
        defaultValue={[]}
        value={[]}
        cancelText="取消"
        okText="确定"
        visible
        dataSource={District}
        onChange={onChange}
        onChangeValidate={onChangeValidate}
      />,
    );

    // popupWrapper = mount(wrapper.instance().getComponent());

    wrapper
      .find('.za-stack-picker__group')
      .at(0)
      .find('.za-stack-picker__stack-column-wrapper')
      .at(0)
      .find('.za-stack-picker__stack-column-item')
      .at(0)
      .simulate('click');

    // 目前不能取到 useState 的值
    expect(onChange).toHaveBeenCalledWith(['340000']);
  });
});

// describe('StackPicker props disabled', () => {
//   let wrapper;
//   const setState = jest.fn();
//   const useStateSpy = jest.spyOn(React, 'useState');
//   useStateSpy.mockImplementation((init) => [init, setState]);
//
//   const onCancel = jest.fn();
//   const onOk = jest.fn();
//
//   beforeEach(() => {
//     wrapper = shallow(
//       <StackPicker
//         visible
//         disabled
//         dataSource={District}
//         onCancel={onCancel}
//         onOk={onOk}
//       />,
//     );
//   });
//
//   it('handle props disabled', () => {
//     wrapper.find('.za-stack-picker__cancel').simulate('click');
//     // 目前不能取到 useState 的值
//     expect(onCancel).toBeCalled();
//   });
// });

describe('StackPicker error type', () => {
  const consoleSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

  afterAll(() => {
    consoleSpy.mockRestore();
  });

  it('handle props error type', () => {
    const onCancel = 1 as any;
    const onOk = 1 as any;
    const displayRender = 1 as any;
    const onChangeValidate = 1 as any;
    const onChange = 1 as any;

    const wrapper = mount(
      <StackPicker
        defaultValue={[]}
        value={[]}
        cancelText="取消"
        okText="确定"
        dataSource={District}
        onCancel={onCancel}
        onOk={onOk}
        displayRender={displayRender}
        onChange={onChange}
        onChangeValidate={onChangeValidate}
      />,
    );

    wrapper.setProps({ visible: true });
    wrapper.update();

    expect(consoleSpy).toHaveBeenCalledWith('displayRender need a function');

    wrapper.find('.za-stack-picker__cancel').simulate('click');
    expect(consoleSpy).toHaveBeenCalledWith('onCancel need a function');

    wrapper.find('.za-stack-picker__submit').simulate('click');
    expect(consoleSpy).toHaveBeenCalledWith('onOk need a function');

    wrapper
      .find('.za-stack-picker__group')
      .at(0)
      .find('.za-stack-picker__stack-column-wrapper')
      .at(0)
      .find('.za-stack-picker__stack-column-item')
      .at(0)
      .simulate('click');
    expect(consoleSpy).toHaveBeenCalledWith('onChangeValidate need a function');
    expect(consoleSpy).toHaveBeenCalledWith('onChange need a function');

    consoleSpy.mockRestore();
  });
});
