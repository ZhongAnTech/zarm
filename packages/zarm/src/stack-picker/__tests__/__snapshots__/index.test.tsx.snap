// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackPicker renders correctly 1`] = `
<ForwardRef(forwardRef)
  cancelText="取消"
  dataSource={
    Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "children": Array [],
                "label": "大观区",
                "value": "340803",
              },
              Object {
                "children": Array [],
                "label": "怀宁县",
                "value": "340822",
              },
              Object {
                "children": Array [],
                "label": "其它区",
                "value": "340882",
              },
            ],
            "label": "安庆市",
            "value": "340800",
          },
        ],
        "label": "安徽省",
        "value": "340000",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "children": Array [],
                "label": "宝山区",
                "value": "310113",
              },
              Object {
                "children": Array [],
                "label": "长宁区",
                "value": "310105",
              },
              Object {
                "children": Array [],
                "label": "崇明县",
                "value": "310230",
              },
              Object {
                "children": Array [],
                "label": "川沙区",
                "value": "310152",
              },
            ],
            "label": "上海市",
            "value": "310100",
          },
        ],
        "label": "上海",
        "value": "310000",
      },
    ]
  }
  defaultValue={Array []}
  itemRender={[Function]}
  okText="确定"
  onChange={[MockFunction]}
  value={Array []}
  valueMember="value"
  visible={true}
>
  <ConfigReceiver
    cancelText="取消"
    dataSource={
      Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "label": "大观区",
                  "value": "340803",
                },
                Object {
                  "children": Array [],
                  "label": "怀宁县",
                  "value": "340822",
                },
                Object {
                  "children": Array [],
                  "label": "其它区",
                  "value": "340882",
                },
              ],
              "label": "安庆市",
              "value": "340800",
            },
          ],
          "label": "安徽省",
          "value": "340000",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "children": Array [],
                  "label": "宝山区",
                  "value": "310113",
                },
                Object {
                  "children": Array [],
                  "label": "长宁区",
                  "value": "310105",
                },
                Object {
                  "children": Array [],
                  "label": "崇明县",
                  "value": "310230",
                },
                Object {
                  "children": Array [],
                  "label": "川沙区",
                  "value": "310152",
                },
              ],
              "label": "上海市",
              "value": "310100",
            },
          ],
          "label": "上海",
          "value": "310000",
        },
      ]
    }
    defaultValue={Array []}
    forwardedRef={null}
    itemRender={[Function]}
    okText="确定"
    onChange={[MockFunction]}
    value={Array []}
    valueMember="value"
    visible={true}
  >
    <Memo()
      cancelText="取消"
      dataSource={
        Array [
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "children": Array [],
                    "label": "大观区",
                    "value": "340803",
                  },
                  Object {
                    "children": Array [],
                    "label": "怀宁县",
                    "value": "340822",
                  },
                  Object {
                    "children": Array [],
                    "label": "其它区",
                    "value": "340882",
                  },
                ],
                "label": "安庆市",
                "value": "340800",
              },
            ],
            "label": "安徽省",
            "value": "340000",
          },
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "children": Array [],
                    "label": "宝山区",
                    "value": "310113",
                  },
                  Object {
                    "children": Array [],
                    "label": "长宁区",
                    "value": "310105",
                  },
                  Object {
                    "children": Array [],
                    "label": "崇明县",
                    "value": "310230",
                  },
                  Object {
                    "children": Array [],
                    "label": "川沙区",
                    "value": "310152",
                  },
                ],
                "label": "上海市",
                "value": "310100",
              },
            ],
            "label": "上海",
            "value": "310000",
          },
        ]
      }
      defaultValue={Array []}
      itemRender={[Function]}
      locale={
        Object {
          "cancelText": "取消",
          "okText": "确定",
        }
      }
      localeCode="zh-CN"
      okText="确定"
      onChange={[MockFunction]}
      value={Array []}
      valueMember="value"
      visible={true}
    >
      <ForwardRef(ForwardedRefComp)
        direction="bottom"
        onMaskClick={[Function]}
        visible={true}
      >
        <WarnIfDeprecatedComp
          direction="bottom"
          forwardedRef={null}
          onMaskClick={[Function]}
          visible={true}
        >
          <Popup
            destroy={true}
            direction="bottom"
            onMaskClick={[Function]}
            prefixCls="za-popup"
            visible={true}
          >
            <Portal
              animationDuration={200}
              animationType="fade"
              destroy={true}
              direction="bottom"
              handlePortalUnmount={[Function]}
              mask={true}
              maskType="normal"
              onMaskClick={[Function]}
              prefixCls="za-popup"
              visible={true}
            >
              <Trigger
                disabled={false}
                onClose={[Function]}
                visible={true}
              >
                <Portal
                  containerInfo={
                    <div
                      class="za-popup-container"
                    >
                      <div
                        class="za-mask za-fade-enter za-mask--normal"
                        style="animation-duration: 200ms;"
                      />
                      <div
                        class="za-popup__wrapper"
                        data-width="0"
                        role="dialog"
                      >
                        <div
                          class="za-popup za-popup--bottom za-popup--show"
                          data-width="0"
                          role="document"
                          style="transition-duration: 200ms; transition-property: transform;"
                        >
                          <div
                            class="za-stack-picker"
                          >
                            <div
                              class="za-stack-picker__container"
                            >
                              <div
                                class="za-stack-picker__wrapper"
                              >
                                <div
                                  class="za-stack-picker__header"
                                >
                                  <div
                                    class="za-stack-picker__cancel"
                                  >
                                    取消
                                  </div>
                                  <div
                                    class="za-stack-picker__title"
                                  >
                                    请选择
                                  </div>
                                  <div
                                    class="za-stack-picker__submit"
                                  >
                                    确定
                                  </div>
                                </div>
                                <div
                                  class="za-stack-picker__crumbs"
                                >
                                  <p>
                                    选择：
                                    
                                  </p>
                                  
                                </div>
                                <div
                                  class="za-stack-picker__group"
                                >
                                  <div
                                    class="za-stack-picker__stack-column za-stack-picker__stack-column--hidden"
                                  >
                                    <div
                                      class="za-stack-picker__stack-column-wrapper"
                                    >
                                      <div
                                        class="za-stack-picker__stack-column-item"
                                      >
                                        安徽省
                                      </div>
                                      <div
                                        class="za-stack-picker__stack-column-item"
                                      >
                                        上海
                                      </div>
                                    </div>
                                  </div>
                                  <div
                                    class="za-stack-picker__stack-column za-stack-picker__stack-column--hidden"
                                  >
                                    <div
                                      class="za-stack-picker__stack-column-wrapper"
                                    >
                                      <div
                                        class="za-stack-picker__stack-column-item"
                                      >
                                        安庆市
                                      </div>
                                    </div>
                                  </div>
                                  <div
                                    class="za-stack-picker__stack-column za-stack-picker__stack-column--hidden"
                                  >
                                    <div
                                      class="za-stack-picker__stack-column-wrapper"
                                    >
                                      <div
                                        class="za-stack-picker__stack-column-item"
                                      >
                                        大观区
                                      </div>
                                      <div
                                        class="za-stack-picker__stack-column-item"
                                      >
                                        怀宁县
                                      </div>
                                      <div
                                        class="za-stack-picker__stack-column-item"
                                      >
                                        其它区
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  }
                >
                  <Mask
                    className="za-fade-enter"
                    prefixCls="za-mask"
                    style={
                      Object {
                        "WebkitAnimationDuration": "200ms",
                        "animationDuration": "200ms",
                      }
                    }
                    type="normal"
                    visible={true}
                  >
                    <div
                      className="za-mask za-fade-enter za-mask--normal"
                      style={
                        Object {
                          "WebkitAnimationDuration": "200ms",
                          "animationDuration": "200ms",
                        }
                      }
                    />
                  </Mask>
                  <div
                    className="za-popup__wrapper"
                    data-width={0}
                    onClick={[Function]}
                    role="dialog"
                    style={Object {}}
                  >
                    <div
                      className="za-popup za-popup--bottom"
                      data-width={0}
                      role="document"
                      style={
                        Object {
                          "WebkitTransitionDuration": "200ms",
                          "WebkitTransitionProperty": "transform",
                          "transitionDuration": "200ms",
                          "transitionProperty": "transform",
                        }
                      }
                    >
                      <div
                        className="za-stack-picker"
                      >
                        <div
                          className="za-stack-picker__container"
                        >
                          <div
                            className="za-stack-picker__wrapper"
                          >
                            <Component
                              cancelText="取消"
                              handleCancel={[Function]}
                              handleOk={[Function]}
                              locale={
                                Object {
                                  "cancelText": "取消",
                                  "okText": "确定",
                                }
                              }
                              okText="确定"
                              prefixCls="za-stack-picker"
                              title="请选择"
                            >
                              <div
                                className="za-stack-picker__header"
                              >
                                <div
                                  className="za-stack-picker__cancel"
                                  onClick={[Function]}
                                >
                                  取消
                                </div>
                                <div
                                  className="za-stack-picker__title"
                                >
                                  请选择
                                </div>
                                <div
                                  className="za-stack-picker__submit"
                                  onClick={[Function]}
                                >
                                  确定
                                </div>
                              </div>
                            </Component>
                            <div
                              className="za-stack-picker__crumbs"
                            >
                              <p>
                                选择：
                              </p>
                            </div>
                            <div
                              className="za-stack-picker__group"
                            >
                              <div
                                className="za-stack-picker__stack-column za-stack-picker__stack-column--hidden"
                                key="0"
                                onClick={[Function]}
                              >
                                <div
                                  className="za-stack-picker__stack-column-wrapper"
                                  onClick={[Function]}
                                >
                                  <div
                                    className="za-stack-picker__stack-column-item"
                                    key="安徽省0"
                                    onClick={[Function]}
                                  >
                                    安徽省
                                  </div>
                                  <div
                                    className="za-stack-picker__stack-column-item"
                                    key="上海1"
                                    onClick={[Function]}
                                  >
                                    上海
                                  </div>
                                </div>
                              </div>
                              <div
                                className="za-stack-picker__stack-column za-stack-picker__stack-column--hidden"
                                key="1"
                                onClick={[Function]}
                              >
                                <div
                                  className="za-stack-picker__stack-column-wrapper"
                                  onClick={[Function]}
                                >
                                  <div
                                    className="za-stack-picker__stack-column-item"
                                    key="安庆市0"
                                    onClick={[Function]}
                                  >
                                    安庆市
                                  </div>
                                </div>
                              </div>
                              <div
                                className="za-stack-picker__stack-column za-stack-picker__stack-column--hidden"
                                key="2"
                                onClick={[Function]}
                              >
                                <div
                                  className="za-stack-picker__stack-column-wrapper"
                                  onClick={[Function]}
                                >
                                  <div
                                    className="za-stack-picker__stack-column-item"
                                    key="大观区0"
                                    onClick={[Function]}
                                  >
                                    大观区
                                  </div>
                                  <div
                                    className="za-stack-picker__stack-column-item"
                                    key="怀宁县1"
                                    onClick={[Function]}
                                  >
                                    怀宁县
                                  </div>
                                  <div
                                    className="za-stack-picker__stack-column-item"
                                    key="其它区2"
                                    onClick={[Function]}
                                  >
                                    其它区
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </Portal>
              </Trigger>
            </Portal>
          </Popup>
        </WarnIfDeprecatedComp>
      </ForwardRef(ForwardedRefComp)>
    </Memo()>
  </ConfigReceiver>
</ForwardRef(forwardRef)>
`;
